packages/core/src/dispatcher.ts

```diff
@@ -14,10 +14,12 @@ import {
 import { Pubsub } from './pubsub/pubsub'
 import { Subscription } from 'rxjs'
 import { MessageBus } from './pubsub/message-bus'
+import { LRUMap } from 'lru_map'
 
 const IPFS_GET_TIMEOUT = 60000 // 1 minute
 const IPFS_MAX_RECORD_SIZE = 256000 // 256 KB
 const IPFS_RESUBSCRIBE_INTERVAL_DELAY = 1000 * 15 // 15 sec
+const IPFS_CACHE_SIZE = 1000000 // 1 million records
```

Comment:
This seems pretty large.  IPFS_MAX_RECORD_SIZE is 256kb, so 1 million records could be up to 256GB just for this cache.  I'd probably start this at like 1000 records, that should be enough for us to be able to load a small handful of streams concurrently and keep all their commits in the cache for normal-case mid-sized streams.  Loading lots of really large streams concurrently risks exhausting the cache, but then we're still no worse off then we are today.

Reply:
Whoops! For some reason I thought the max would end up being 256MB 😬

1000 records sounds reasonable.